sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TestMethodShouldHaveCorrectSignature.MsTest.cs

```diff
@@ -44,6 +45,59 @@ class MsTestTest
         [TestMethod]
         public async Task DoSomethingAsync() // Compliant
         {
+            return;
+        }
+    }
+}
+
+namespace Tests.Diagnostics
+{
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System.Threading.Tasks;
+
+    [TestClass]
+    public class MsTestTest_DataTestMethods
```

Comment:
MSTest - same test cases, but repeated for the _DataTestMethod_ attribute.

The tests for the other frameworks follow the same pattern - duplicating the existing test cases for the other attributes.